pkg/server/server.go

```diff
@@ -339,8 +391,29 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 		graph.WithResolveNodeBreadthLimit(s.resolveNodeBreadthLimit),
 	)
 
-	cycleDetectionCheckResolver.SetDelegate(localChecker)
-	localChecker.SetDelegate(cycleDetectionCheckResolver)
+	if s.dispatchThrottlingCheckResolverEnabled {
+		dispatchThrottlingConfig := graph.DispatchThrottlingCheckResolverConfig{
+			TimerTickerFrequency: s.dispatchThrottlingCheckResolverTimerTickerFrequency,
+			Level:                s.dispatchThrottlingCheckResolverLevel,
+			Rate:                 s.dispatchThrottlingCheckResolverRate,
+		}
+
+		s.logger.Info("Enabling dispatch throttling",
+			zap.Duration("TimerTickerFrequency", s.dispatchThrottlingCheckResolverTimerTickerFrequency),
+			zap.Uint32("LowPriorityLevel", s.dispatchThrottlingCheckResolverLevel),
+			zap.Uint32("LowPriorityRate", s.dispatchThrottlingCheckResolverRate),
+		)
+
+		dispatchThrottlingCheckResolver := graph.NewDispatchThrottlingCheckResolver(dispatchThrottlingConfig)
+		dispatchThrottlingCheckResolver.SetDelegate(localChecker)
+		s.dispatchThrottlingCheckResolver = dispatchThrottlingCheckResolver
+
+		cycleDetectionCheckResolver.SetDelegate(dispatchThrottlingCheckResolver)
+		localChecker.SetDelegate(cycleDetectionCheckResolver)
+	} else {
```

Comment:
This else is redunadnt as these delegate assignments remain unchanged and are the defaults if dispatch throttling is disabled. I recommend an alternative setup that looks like this, for example:

```go
cycleDetectionCheckResolver := graph.NewCycleDetectionCheckResolver()
s.checkResolver = cycleDetectionCheckResolver

localChecker := graph.NewLocalChecker(
	graph.WithResolveNodeBreadthLimit(s.resolveNodeBreadthLimit),
)

cycleDetectionCheckResolver.SetDelegate(localChecker)
localChecker.SetDelegate(cycleDetectionCheckResolver)

var dispatchThrottlingCheckResolver *graph.DispatchThrottlingCheckResolver
if s.dispatchThrottlingCheckResolverEnabled {
    ...

	dispatchThrottlingCheckResolver = graph.NewDispatchThrottlingCheckResolver(...)
	dispatchThrottlingCheckResolver.SetDelegate(localChecker)
	s.dispatchThrottlingCheckResolver = dispatchThrottlingCheckResolver

	cycleDetectionCheckResolver.SetDelegate(dispatchThrottlingCheckResolver)
}

if s.checkQueryCacheEnabled {
	...

	cachedCheckResolver := graph.NewCachedCheckResolver(...)
	s.cachedCheckResolver = cachedCheckResolver

	cachedCheckResolver.SetDelegate(localChecker)

	if dispatchThrottlingCheckResolver != nil {
		dispatchThrottlingCheckResolver.SetDelegate(cachedCheckResolver)
	} else {
		cycleDetectionCheckResolver.SetDelegate(cachedCheckResolver)
	}
}
```